NewForce-Cohort-7/tabloidfullstack-agnostic-pillows
# Ticket #37 Upload a Post Header Image

# Description
This is on the `nm-uploadpostheader` branch.

For your reference: [SQL Data](https://github.com/NewForce-Cohort-7/tabloidfullstack-agnostic-pillows/tree/main/SQL)

**# Title** in Trello - [Ticket #37 Upload a Post Header Image](https://trello.com/c/eF8iNK6p/38-37-upload-post-header-image)
Per the ticket: As an author I would like to be able to update a header image for a Post so that I do not have to find an image online to link to.

This functionality has now been implemented. 

## Type of change

# Testing Instructions

**GIT this branch:**
1. Clone this repository to your own machine using `git clone https://github.com/NewForce-Cohort-7/tabloidfullstack-agnostic-pillows.git`
2. `git fetch -a` in the master branch to retrieve this branch named `branch-name`.
3.  checkout this branch using the command `git checkout branch-name`


**Create the database:**
1. Open Visual Studio.
2. At the start screen select "Open a Local Folder"
4. Navigate to your workspace directory, open the `tabloidfullstack-agnostic-pillows` folder, then select the SQL folder
5. Run both .sql files in the [SQL folder](https://github.com/NewForce-Cohort-7/tabloidfullstack-agnostic-pillows/tree/main/SQL) (The first file creates the database and tables if they don’t already exist, and the second file inserts a very small amount of sample data into those tables for testing.).

**Run the app:**
1. Open Visual Studio
2. At the start screen select Open a project or solution.
3. Navigate to your workspace directory, open the `tabloidfullstack-agnostic-pillows` folder, and select the .sln file to run the solution.
4. Hit the Green play button on the toolbar that says TabloidFullStack.
5. This will launch the Swagger page in your default browser. The Swagger page provides the endpoints and data from the SQL server that we created. 
6. Now we need to run the React side of the app. 
7. In your command line, cd into the `client` folder and into the `tabloid` folder. 
8. Start by running `npm install react-scripts`. Once this has been installed, type `npm start`. 
9. This will open the React app in your default browser. You should now see the log-in page. 
10. Log in to user `agrzeskowski3@fc2.comx`. You do not need to enter a password. 

**Test the new feature:**
Two different implementations of this feature were included. One being when editing one of a users posts, and the other when creating a brand new post.

1. Navigate to the My Posts option in the header.
2. Click on any of the posts that populate on the screen.
3. Click the Edit Post button on the bottom left of the card.
4. The Image Url option has changed to be a file upload. Choose an image file from your local files.
5. After you pick the file, you should confirm that the name of the file you picked now populates the file input.
6. Save the edit, this will redirect you back to the Post Details page, now displaying the newly uploaded image file.

The implementation and step to test this for the Create Post option follow the same general steps.

1. Navigate to the Posts page.
2. Click the Create New Post button at the top left of the screen.
3. Fill the form out as you would, this time instead of copying an image from the internet, use the File Upload to choose an image file from your local files.
4. Finish creating the post, this will redirect you back to the Post Details page, now displaying the newly uploaded image file in your new post.

# Checklist:
